### PR TITLE
[base] cleanups

### PR DESCRIPTION
**Pull request by Dushistov**
basically removing deprecated code

### CODE REVIEW

File: base/string_utils.hpp [hunk 1]
```diff
@@ -584,49 +584,11 @@ int UpperBoundOnChars()
...
-  char * end = buf + sz;
-  char * beg = to_string_digits(end, i);
-  return std::string(beg, end - beg);
-}
 }  // namespace impl
 
-inline std::string to_string(int32_t i) { return impl::to_string_signed(i); }
-inline std::string to_string(int64_t i) { return impl::to_string_signed(i); }
-inline std::string to_string(uint64_t i) { return impl::to_string_unsigned(i); }
+inline std::string to_string(int32_t i) { return std::to_string(i); }
```
Comment (biodranik) [inline base/string_utils.hpp:589]: 1. How fast is std::to_string compared to the previous implementation? Here are some numbers: https://www.zverovich.net/2013/09/07/integer-to-string-conversion-in-cplusplus.html
2. If performance is ok, I would prefer to delete strings::to_string completely and replace it everywhere in the code with std::to_string.

File: base/threaded_list.hpp [hunk 2]
```diff
@@ -12,9 +12,28 @@ class ThreadedList : public ThreadedContainer
...
+
+    while ((m_isEmpty = m_list.empty()))
+    {
+      if (IsCancelled())
+        break;
+
+      if (doFirstWait)
+        doFirstWait = false;
+
+      m_Cond.wait(lock);
```
Comment (biodranik) [inline base/threaded_list.hpp:29]: s/m_Cond/m_cond/g

File: base/threaded_list.hpp [hunk 3]
```diff
@@ -12,9 +12,28 @@ class ThreadedList : public ThreadedContainer
...
+  {
+    bool doFirstWait = true;
+
+    while ((m_isEmpty = m_list.empty()))
+    {
+      if (IsCancelled())
+        break;
+
+      if (doFirstWait)
+        doFirstWait = false;
```
Comment (biodranik) [inline base/threaded_list.hpp:27]: What is it used for?

File: base/string_utils.hpp (hunk 1 above)
Comment (Dushistov) [reply base/string_utils.hpp:569]: > 1. How fast is std::to_string compared to the previous implementation?

Looking at source code it is definitely faster. Current `string_utils.hpp` implementation is just naive one,
the implementation in libstdc+++ and libc++ is similar to Alexandrescu's idea mentioned in your [link](https://www.zverovich.net/2013/09/07/integer-to-string-conversion-in-cplusplus.html). So on graph it should be near `fmt::format_int`

> If performance is ok, I would prefer to delete strings::to_string completely and replace it everywhere in the code with std::to_string

Not sure what do you mean here. I already replaced all occurrences. The `base::to_string` != `std::to_string`. `std::to_string` is implemented for numbers only, the `base::to_string` for any types. I replaced part specialized implemented for numbers, `std::to_string` can replace other part.

File: base/threaded_list.hpp (hunk 2 above)
Comment (Dushistov) [reply base/threaded_list.hpp:29]: why? this is already existing data member, I do not invent this name.

File: base/threaded_list.hpp (hunk 3 above)
Comment (Dushistov) [reply base/threaded_list.hpp]: Have no idea. May have been used before for debugging, but has not been removed. 
I removed "extra" code from this function.

Comment (biodranik) [reply base/threaded_list.hpp]: If you already touched the code, then make it better ;-)

File: base/threaded_list.hpp (hunk 2 above)
Comment (biodranik) [reply base/threaded_list.hpp:24]: To fix invalid variable naming and make the code better and more consistent.

File: base/shared_buffer_manager.cpp [hunk 4]
```diff
@@ -9,13 +9,13 @@ SharedBufferManager & SharedBufferManager::instance()
 
 void SharedBufferManager::clearReserved()
 {
-  threads::MutexGuard g(m_mutex);
+  std::unique_lock<std::mutex> g(m_mutex);
```
Comment (vng) [inline base/shared_buffer_manager.cpp:12]: std::lock_guard is better and more informative here and below. We don't share or transfer lock here, right?

File: base/threaded_list.hpp [hunk 5]
```diff
@@ -12,9 +12,20 @@ class ThreadedList : public ThreadedContainer
 private:
 
   std::list<T> m_list;
-  bool m_isEmpty;
-  std::string m_resName;
+  std::atomic<bool> m_isEmpty;
 
+  bool WaitNonEmpty(std::unique_lock<std::mutex> &lock)
+  {
+    while ((m_isEmpty = m_list.empty()))
```
Comment (vng) [inline base/threaded_list.hpp:19]: Why 2 pair of braces?

**Review by vng — CHANGES_REQUESTED** [review]

**Comment by vng** [comment]
iphone/Maps/Maps.xcodeproj files should be updated manually (?!).

**Comment by biodranik** [comment]
Please also delete lines with removed files from any xcode project files in xcode/ folder and subfolders.

**Comment by biodranik** [comment]
LGTM from me after removing deleted files from xcode project file.

**Comment by vng** [comment]
@Dushistov I made force push into your branch if you don't mind :)

**Pull request merged**